sqlserver/tests/test_statements.py

```diff
@@ -452,9 +452,7 @@ def test_statement_metadata(
 ):
     check = SQLServer(CHECK_NAME, {}, [dbm_instance])
 
-    query = '''
-    -- Test comment
-    select * from sys.databases'''
+    query = 'select * from sys.databases'
```

Comment:
### fix reason

When there is a cache of queries with the same `query_hash` and `query_plan_hash`, [this max(plan_handle_and_offsets)](https://github.com/DataDog/integrations-core/blob/adf5d2a523ca41945747eb858c092a0d39523b01/sqlserver/datadog_checks/sqlserver/statements.py#L68) may be different from before.

(1) and (2) are the same `query_hash` and `query_plan_hash`, and (2) was used preferentially before this fix.

(1)
```sql
-- Test comment
select * from sys.databases;
```

(2)
```sql
select * from sys.databases;
```

Because (2) is used preferentially in the test case, comments were not set on the following code.
https://github.com/DataDog/integrations-core/blob/a14f8ece045f7105dcb61b1f18e8ee3f35ecf05b/sqlserver/datadog_checks/sqlserver/statements.py#L310

As a result, the following code used metadata to set the comments

https://github.com/DataDog/integrations-core/blob/a14f8ece045f7105dcb61b1f18e8ee3f35ecf05b/sqlserver/datadog_checks/sqlserver/statements.py#L326-L327


On the other hand, (1) is used preferentially in the test case after fix, and `Test comment` set on the following code.
https://github.com/DataDog/integrations-core/blob/a14f8ece045f7105dcb61b1f18e8ee3f35ecf05b/sqlserver/datadog_checks/sqlserver/statements.py#L310

As a result, the following code didn't passed and failed assert with metadata.
https://github.com/DataDog/integrations-core/blob/a14f8ece045f7105dcb61b1f18e8ee3f35ecf05b/sqlserver/datadog_checks/sqlserver/statements.py#L326-L327